hs_core/views/utils.py

```diff
@@ -775,9 +775,56 @@ def move_or_rename_file_or_folder(user, res_id, src_path, tgt_path, validate_mov
 
     # ensure the target_full_path contains the file name to be moved or renamed to
     # if we are moving to a directory, put the filename into the request.
+    # TODO: #2105: this is not a sufficient check that the move is sane.
+    # TODO: #2105: separate the two use cases in this situation to avoid confusion
     if src_file_dir != tgt_file_dir and tgt_file_name != src_file_name:
         tgt_full_path = os.path.join(tgt_full_path, src_file_name)
 
+    # TODO: why is this optional?
+    if validate_move_rename:
+        # this must raise ValidationError if move/rename is not allowed by specific resource type
+        if not resource.supports_rename_path(src_full_path, tgt_full_path):
+            raise ValidationError("File/folder move/rename is not allowed.")
+
+    istorage.moveFile(src_full_path, tgt_full_path)
+
+    rename_irods_file_or_folder_in_django(resource, src_full_path, tgt_full_path)
+
+    hydroshare.utils.resource_modified(resource, user, overwrite_bag=False)
+
+
+# TODO: modify this to take short paths not including data/contents
+def move_to_folder(user, res_id, src_path, tgt_path, validate_move_rename=True):
+    """
+    Move or rename a file or folder in hydroshareZone or any federated zone used for HydroShare
+    resource backend store.
+    :param user: requesting user
+    :param res_id: resource uuid
+    :param src_path: the relative paths for the source file or folder under res_id collection
+    :param tgt_path: the relative paths for the target folder under res_id collection
+    :param validate_move_rename: if True, then only ask resource type to check if this action is
+            allowed. Sometimes resource types internally want to take this action but disallow
+            this action by a user. In that case resource types set this parameter to False to allow
+            this action.
+    :return:
+
+    Note: this utilizes partly qualified pathnames data/contents/foo rather than just 'foo'
+    """
+    if __debug__:
+        assert(src_path.startswith("data/contents/"))
+        assert(tgt_path.startswith("data/contents/"))
+
+    resource = hydroshare.utils.get_resource_by_shortkey(res_id)
+    istorage = resource.get_irods_storage()
+    src_full_path = os.path.join(resource.root_path, src_path)
+    tgt_full_path = os.path.join(resource.root_path, tgt_path)
+
+    # tgt_file_name = os.path.basename(tgt_full_path)
+    # tgt_file_dir = os.path.dirname(tgt_full_path)
+    # src_file_name = os.path.basename(src_full_path)
+    # src_file_dir = os.path.dirname(src_full_path)
```

Comment:
@hyi The way you conflated move and rename made it necessary for the UI to know whether the endpoint was a file or folder. The UI didn't have enough information and made a naïve choice. That is the code that was causing the dot error. The easiest way to eliminate that line is to delegate that logic to the REST API rather than leaving it on the back end.